### PR TITLE
Initial commit of contribution graph for community review

### PR DESCRIPTION
**Pull request by justinrassier**
Here is my first stab at a reusable module in Elm. Feel free to offer any insight on how to do things better, from syntax to design patterns! A couple things I know that need to be figured out/cleaned up before this would actually be published 

* Elm docs - I started to play around with the document format, but the docs are far from complete
* CSS - My internal project relied on a stylesheet being available. I need to figure out how to encapsulate those styles in a reusable way. I included a slimmed down stylesheet you can put into your project to try it out to at least see it in action. It is far from complete.
* The concept of "points" is probably a bit too use case specific. I probably want to remove that as part of this if I was to make it more widely useful 
* Tests - I have tests for the date calculations, they were just part of may main application before I split this out. I would put them back for sure!

### CODE REVIEW

File: src/DateHelpers.elm [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+        Nov ->
+            9
+
+        Dec ->
+            10
+
+
+monthToString : Month -> String
+monthToString month =
```
Comment (groteck) [inline src/DateHelpers.elm]: You can simplify `monthToStandardValue << String.fromInt` reusing the prev function

Comment (justinrassier) [reply src/DateHelpers.elm]: Wow, can't believe I missed that! Thanks!

**Review by yarus23 — COMMENTED** [review]
It is possible to convert many similar expressions 
 `viewSvgDay year (doy Sun) (getContributionsForDayOfYear timeZone year (doy Sun) contributions)`

to 
``` 
List.range 0 6
|> List.map (\ i ->
let
  doy = i + 1 + weekStartDay
in
   viewSvgDay year doy (getContributionsForDayOfYear timeZone year doy contributions)) 
```

and remove previous function doy

**Comment by justinrassier** [comment]
@yarus23 You are right, that part of the code felt clumsy, thanks for the idea to go revisit it!

**Pull request merged**